### PR TITLE
Adding tag functionality to component config

### PR DESCRIPTION
**Pull request by xaviemirmon**

### CODE REVIEW

File: packages/core/components/ComponentList/index.tsx [hunk 1]
```diff
@@ -5,10 +5,24 @@ import { Draggable } from "react-beautiful-dnd";
...
 import { Grid } from "react-feather";
+import { Heading } from "../Heading";
 
 const getClassName = getClassNameFactory("ComponentList", styles);
 
 export const ComponentList = ({ config }: { config: Config }) => {
+  const groupedComponents = {};
+  const defaultTagLabel = config?.tags?.defaultLabel
+    ? config.tags.defaultLabel
+    : "Uncategorised";
```
Comment (chrisvxd) [inline packages/core/components/ComponentList/index.tsx:16]: Can we keep "Components" for the default label here?

File: packages/core/types/Config.tsx [hunk 2]
```diff
@@ -86,6 +87,9 @@ export type Config<
       Props[ComponentName]
     >;
   };
+  tags?: {
+    defaultLabel?: string;
+  };
```
Comment (chrisvxd) [inline packages/core/types/Config.tsx:92]: I think we can make better use of this API-namespace to future proof it

```
  tags: {
    default: {
      label: "Others"
    },
    layout: {
      label: "Layout"
    }
  }
```

We can then create a strict type-link between the tags specified here and the tags available for the components

File: packages/core/components/ComponentList/index.tsx [hunk 3]
```diff
@@ -17,51 +31,67 @@ export const ComponentList = ({ config }: { config: Config }) => {
...
+          {Object.entries(groupedComponents)
+            .sort()
+            .map(([tag, components]) => {
+              return (
+                <>
+                  {Object.keys(groupedComponents).length > 1 ? (
+                    <Heading rank={3} size={"xs"}>
+                      {tag}
+                    </Heading>
+                  ) : null}
```
Comment (chrisvxd) [inline packages/core/components/ComponentList/index.tsx:43]: I'm not totally sure that using Headings like this is the way to go for the UI here.

It also exacerbates some existing visual issues when cancelling a drag


https://github.com/measuredco/puck/assets/985961/5590e311-029e-42e5-a6db-4c42d2c7e627



I wonder if we should render multiple

**Review by chrisvxd — COMMENTED** [review]
Thanks @xaviemirmon! This is an interesting proposal

#87 wasn't quite ready for picking up yet (we haven't marked it as "Ready" yet, still in "Triage" - we haven't got contribution guidelines yet so that's probably not clear). It's also marked with "opinions wanted" because we were keen to have some dialog before setting on a direction.

This PR implements option 2 in the proposals, which is valid, but I'd like to also discuss the other options available to us before we finalise this approach.

Regardless, this is still really useful input that can help with deciding the best path forwards. Let's pick up conversation in #87 and discuss what option 1 and 3 might also look like.

If we go with option 2, then we can just pick this PR up. I've given my feedback where appropriate!

File: packages/core/types/Config.tsx (hunk 2 above)
Comment (xaviemirmon) [reply packages/core/types/Config.tsx:92]: Yeah, I like it. I wonder if it's also worth adding a weight field like so

` tags: {
    default: {
      label: "Others"
      weight: 0
    },
    layout: {
      label: "Layout"
      weight: 2
    }
  }`
  
  That way you could reorder the order that they display?

**Comment by chrisvxd** [comment]
Converted to draft as we discuss #87

**Comment by chrisvxd** [comment]
Closing in favour of #183

**Pull request closed without merging**